decimal/decimal.nim

```diff
@@ -6,3 +6,93 @@
 # at your option. This file may not be copied, modified, or distributed except according to those terms.
 
 import decimal_lowlevel
+
+type 
+    DecimalType = ptr mpd_t
+
+const
+    DEFAULT_PREC = MPD_RDIGITS * 2
+    DEFAULT_EMAX = when (sizeof(int) == 8): 999999999999999999 else: 425000000
+    DEFAULT_EMIN = when (sizeof(int) == 8): -999999999999999999 else: -425000000
+
+var CTX: mpd_context_t
+var CTX_ADDR = addr CTX
+mpd_defaultcontext(CTX_ADDR)
+
+proc setPrec*(prec: int) =
+    if 0 < prec:
+        let success = mpd_qsetprec(CTX_ADDR, prec)
+        if success == 0:
+            echo "Bad prec" 
+
+proc `$`*(s: DecimalType): cstring = mpd_to_sci(s, 0)
+
+proc initDecimal*(): DecimalType = mpd_new(CTX_ADDR)
```

Comment:
The convention in Nim is `initFoo` for value types and `newFoo` for reference types.
This should be renamed newDecimal.

Reply:
fixed :-)